File: activity/app/src/main/java/com/rikkei/training/activity/MainActivity.java
package com.rikkei.training.activity;

import androidx.annotation.NonNull;
import androidx.appcompat.app.AppCompatActivity;

import android.Manifest;
import android.content.Intent;
import android.os.Build;
import android.os.Bundle;
import android.util.Log;
import android.view.View;
import android.widget.Button;

public class MainActivity extends AppCompatActivity implements View.OnClickListener {
    private static final String TAG = MainActivity.class.getSimpleName();
    private Button mButton;
    private MediaManager media;
    private String link = "/storage/emulated/0/Download/Di Du Dua Di - Bich Phuong.mp3";
    private final String[] PERMISSIONS = {
            Manifest.permission.READ_EXTERNAL_STORAGE
    };
    private SystemData data;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        Log.d(TAG, "on Create");
        setContentView(R.layout.activity_main);
        if (checkPermission()) {
            initViews();
        }
    }

    private boolean checkPermission() {
        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
            for (String p : PERMISSIONS) {
                if (checkSelfPermission(p) != getPackageManager().PERMISSION_GRANTED) {
                    requestPermissions(PERMISSIONS, 0);
                    return false;
                }
            }
        }
        return true;
    }

    @Override
    public void onRequestPermissionsResult(int requestCode, @NonNull String[] permissions, @NonNull int[] grantResults) {
        super.onRequestPermissionsResult(requestCode, permissions, grantResults);
        if (checkPermission()) {
            initViews();
        } else {
            finish();
        }
    }

    private void initViews() {
        mButton = findViewById(R.id.btn_act_a);
        mButton.setOnClickListener(this);

        data = new SystemData(this);
        data.readData();
        media = new MediaManager(link,this);
        media.create();
    }

    @Override
    protected void onStart() {
        super.onStart();
        media.start();
        Log.d(TAG, "on Start");
    }

    @Override
    protected void onResume() {
        super.onResume();
        Log.d(TAG, "on Resume");
    }

    @Override
    protected void onPause() {
        super.onPause();
        Log.d(TAG, "on Pause");
    }

    @Override
    protected void onStop() {
        super.onStop();
        Log.d(TAG, "on Stop");
    }

    @Override
    protected void onDestroy() {
        super.onDestroy();
        media.stop();
        Log.d(TAG, "on Destroy");
    }

    @Override
    public void onClick(View view) {
        Intent intent = new Intent(this, AActivity.class);
        media.pause();
        startActivity(intent);
    }
}

File: UIKotlin/app/src/main/java/com/rikkei/training/uikotlin/activity/MainActivity.kt
package com.rikkei.training.uikotlin.activity

import android.content.Intent
import android.view.View
import com.rikkei.training.uikotlin.Adapter
import com.rikkei.training.uikotlin.R
import com.rikkei.training.uikotlin.base.BaseActivity
import com.rikkei.training.uikotlin.model.Image
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : BaseActivity(), View.OnClickListener {
    override fun initAct() {
        data = ArrayList()
        data!!.add(Image(R.drawable.ic_launcher_background))
        data!!.add(Image(R.drawable.ic_launcher_background))
        data!!.add(Image(R.drawable.ic_launcher_background))
        data!!.add(Image(R.drawable.ic_launcher_background))
        data!!.add(Image(R.drawable.ic_launcher_background))
        data!!.add(Image(R.drawable.ic_launcher_background))
        data!!.add(Image(R.drawable.ic_launcher_background))

        adapter = Adapter(data!!, this)
        recycler_linear!!.adapter = adapter
        recycler_grid!!.adapter = adapter
        recycler_staggered!!.adapter = adapter
        recycler_staggered!!.isNestedScrollingEnabled = false
        btn_actToolbar.setOnClickListener(this)
    }

    override fun getLayoutId(): Int {
        return R.layout.activity_main
    }

    override fun onClick(p0: View?) {
        var intent = Intent(this, ToolbarActivity::class.java)
        startActivity(intent)
    }

    private var adapter: Adapter? = null
    private var data: ArrayList<Image>? = null

}
